### PR TITLE
PORT THE AUDIO TO SFML!!!!

### LINKED ISSUE #435: Port Music system to SFML
The music system needs to be ported from SDL_music to SFML’s Audio module.

Cf. http://www.sfml-dev.org/tutorials/2.3/audio-sounds.php

Valete,
Quintus

### PR DESCRIPTION
**Pull request by kirbyfan64**
Closes #435. Any advice on what to do with the two XXX's remaining would be great!

I don't know of anything left that's wrong. Please try it out and let me know of anything you find that doesn't work!

I'll squash this when I'm sure it's done.

### CODE REVIEW

**Comment by kirbyfan64** [comment]
After several more commits, I'm pretty sure I ironed out *every* bug.

**Comment by kirbyfan64** [comment]
Bump? Been two weeks...

File: tsc/src/core/global_basic.hpp [hunk 1]
```diff
@@ -98,10 +98,24 @@
...
-#include <SDL_mixer.h>
 #include <SDL_ttf.h>
 #include <SDL_syswm.h>
 
+// SFML
+/*
+   XXX: This is a hack! X11 defines a macro Status that expands to int, so SFML's
+   Status enum ends up being invalid syntax. Who thought that was a good idea
+   again?
+*/
```
Comment (Quintus) [inline tsc/src/core/global_basic.hpp]: X11 had definitions that already clashed with SDL and CEGUI. Take a look at how global_basic.hpp looked like in the old OpenGL/SDL port and you’ll see there are comments about that: https://github.com/Secretchronicles/TSC/blob/devel/tsc/src/core/global_basic.hpp#L87-L89

The comment only talks about CEGUI, but AFAIK it affects SDL as well. Try moving all the SFML includes before the X11 includes (so they are kept as a block of related statements) and see if that resolves the problem. If not, we’ll have to stick to this hack for now.

**Comment by Quintus** [comment]
I have had a very busy time, and as it appears the project fairly quickly comes to a standstill if I am not available. It’s be nice to have at least one additional dev who takes care of looking after things here. @Luiji?

I find this PR looks okay, no obvious problems with it. I have not yet tested it, but given the current state of the SFML port, it can’t easily be tested anyway. Therefore, I suggest to merge this.

Vale,
Quintus

**Comment by kirbyfan64** [comment]
Stupid. GitHub.

**Pull request closed without merging**